src/fetch.ts

```diff
@@ -9,13 +9,26 @@ import logger from './logger';
 import invariant from './util/invariant';
 import { sleep } from './util/time';
 
+function sanitizeUrl(url: string): string {
+  try {
+    const parsedUrl = new URL(url);
+    if (parsedUrl.username || parsedUrl.password) {
+      parsedUrl.username = '***';
+      parsedUrl.password = '***';
```

Comment:
Quick thought, should we at least show the first/last character of the username and password? 

Could be helpful for debugging.